scala-package/infer/src/main/scala/ml/dmlc/mxnet/infer/ImageClassifier.scala

```diff
@@ -0,0 +1,160 @@
+    val inputBatch = ListBuffer[BufferedImage]()
+    for (imgFile: File <- dir.listFiles()){
+      val img = ImageIO.read(imgFile)
+      inputBatch += img
+    }
+    inputBatch.toList
+  }
+
+  /**
+    * Reshape the input image to new shape
+    *
+    * @param img       image
+    * @param newWidth  rescale to new width
+    * @param newHeight rescale to new height
+    * @return Rescaled BufferedImage
+    */
+  def getScaledImage(img: BufferedImage, newWidth: Int, newHeight: Int): BufferedImage = {
+    val resizedImage = new BufferedImage(newWidth, newHeight, BufferedImage.TYPE_INT_RGB)
+    val g = resizedImage.createGraphics()
+    g.drawImage(img, 0, 0, newWidth, newHeight, null)
+    g.dispose()
+
+    resizedImage
+  }
+
+  /**
+    * Read image file from provided path
+    *
+    * @param resizedImage BufferedImage to get pixels from
+    * @return NDArray pixels array
+    */
+  def getPixelsFromImage(resizedImage: BufferedImage): NDArray = {
+    val w = resizedImage.getWidth
+    val h = resizedImage.getHeight
+
+    val pixels = new ListBuffer[Float]()
+
+    for (x <- 0 until h) {
+      for (y <- 0 until w) {
+        val color = resizedImage.getRGB(y, x)
+        val red = (color & 0xff0000) >> 16
+        val green = (color & 0xff00) >> 8
+        val blue = color & 0xff
+        pixels += red
+        pixels += green
+        pixels += blue
+      }
+    }
+
+    val reshaped_pixels = NDArray.array(pixels.toArray, shape = Shape(224, 224, 3))
+
+    val swapped_axis = NDArray.swapaxes(reshaped_pixels, 0, 2)
+    val pixelsNdarray = NDArray.swapaxes(swapped_axis, 1, 2)
+
+    pixelsNdarray
+  }
+
+  /**
+    * To classify the image according to the provided model
+    *
+    * @param inputImage PathPrefix of the input image
+    * @param topK Get top k elements with maximum probability
+    * @return List of list of tuples of (class, probability)
+    */
+  def classifyImage(inputImage: BufferedImage,
+                        topK: Option[Int] = None): IndexedSeq[List[(String, Float)]] = {
+
+    val width = inputDescriptors(0).shape(2)
+    val height = inputDescriptors(0).shape(3)
+
+    val scaledImage = this.getScaledImage(inputImage, width, height)
+    val pixelsNdarray = this.getPixelsFromImage(scaledImage)
+
+    val input = IndexedSeq(pixelsNdarray.reshape(inputDescriptors(0).shape))
+
+    val output = super.classifyWithNDArray(input, topK)
+
+    IndexedSeq(output(0))
+  }
+
+  /**
+    * To classify batch of input images according to the provided model
+    * @param inputBatch Input batch of Buffered images
+    * @param topK Get top k elements with maximum probability
+    * @return List of list of tuples of (class, probability)
+    */
+  // [TODO] change to batched ndarrays to improve performance
+  def classifyImageBatch(inputBatch: Traversable[BufferedImage], topK: Option[Int] = None):
+  List[List[(String, Float)]] = {
+    val result = ListBuffer[List[(String, Float)]]()
+    for (image <- inputBatch) {
```

Comment:
as discussed offline
the batch of input should be converted to an NDArray and run with ClassifywithNDArray